stores/team_store.jsx

```diff
@@ -368,7 +368,7 @@ class TeamStoreClass extends EventEmitter {
 
     subtractUnread(teamId, msgs, mentions) {
         let member = this.getMyTeamMembers().filter((m) => m.team_id === teamId)[0];
-        if (member) {
+        if (member && (msgs > 0 || mentions > 0)) {
```

Comment:
needs_team component renders again because of the team_store listener.  So adding a check to decrease unnecessary renders.

Reply:
We should do a separate PR to address this. NeedsTeam already has a container, so it should just listen directly to the redux store for the current team instead of needing to change this.

Reply:
I just changed this because of the top-level update on channel switch which I could just avoid by adding this line. If there is a selector which skips the render for this scenario then this won't be needed as well.
![screen shot 2017-12-19 at 10 18 03 pm](https://user-images.githubusercontent.com/4973621/34168481-9de36612-e50a-11e7-916b-2c2b67ece968.png)